### ISSUE TITLE
Add `alces.self` parameter

### ISSUE THREAD

**Issue by ColonelPanicks** [alces-software/metalware#209]
The self node does not appear in `/etc/hosts`, after setting up metalware and configuring a nodes group, the file looks as follows:
```
[root@headnode1 ~]# cat /etc/hosts


########## METALWARE_START ##########
# This section of this file is managed by Alces Metalware. Any changes made to
# this file between the METALWARE_START and METALWARE_END markers may be lost;
# you should make any changes you want to persist outside of this section or to
# the template directly.
10.10.1.1 node01.pri.testcluster.compute node01.pri node01
10.11.1.1 node01.mgt.testcluster.compute node01.mgt
10.12.1.1 node01.ib.testcluster.compute node01.ib
10.11.101.1 node01.bmc.mgt.testcluster.compute node01.bmc

10.10.1.2 node02.pri.testcluster.compute node02.pri node02
10.11.1.2 node02.mgt.testcluster.compute node02.mgt
10.12.1.2 node02.ib.testcluster.compute node02.ib
10.11.101.2 node02.bmc.mgt.testcluster.compute node02.bmc

10.10.1.3 node03.pri.testcluster.compute node03.pri node03
10.11.1.3 node03.mgt.testcluster.compute node03.mgt
10.12.1.3 node03.ib.testcluster.compute node03.ib
10.11.101.3 node03.bmc.mgt.testcluster.compute node03.bmc

10.10.1.4 node04.pri.testcluster.compute node04.pri node04
10.11.1.4 node04.mgt.testcluster.compute node04.mgt
10.12.1.4 node04.ib.testcluster.compute node04.ib
10.11.101.4 node04.bmc.mgt.testcluster.compute node04.bmc

10.10.1.5 node05.pri.testcluster.compute node05.pri node05
10.11.1.5 node05.mgt.testcluster.compute node05.mgt
10.12.1.5 node05.ib.testcluster.compute node05.ib
10.11.101.5 node05.bmc.mgt.testcluster.compute node05.bmc
########## METALWARE_END ##########
```

I suspect this is due to the self node not belonging to any groups so it cannot be accessed via `alces.groups`. Maybe having an `alces.self` or some other way of accessing the self node would be cleaner than hacking it into a group?

**Comment by WilliamMcCumstie** [comment]
I have setup `alces.self` to return the config for self (see branch `feature/alces-self`). However their will need to be a change in `metalware-default`. Also the self answers aren't natively available (all thought they can be accessed from the self config).

Let me know if their is anything else you need and I will merge the branch through and make another release candidate.

**Comment by WilliamMcCumstie** [comment]
Their is a tenative fix in `feature/alce-self` branch. However it is currently causing `bin/metal view-config` to break. This fix will be put into `2017.1.1`

**Comment by bobwhitelock** [comment]
I don't think this is a bug, it's expected behaviour to do with the Metalware content - the `hosts` template states that each node in each configured group should be included in the `hosts` file, so that is what happens.

If we want a node with name `self` to be put into the `hosts` file, can we not just define a group including this node and then it will appear? Alternatively, it can be manually added outside of the `METALWARE_START`/`METALWARE_END` block; I believe this was one reason this feature was originally added.

If we do need to add a new `alces.self` parameter we can do that in a later release as needed, but currently we're just trying to fix existing bugs prior to releasing `2017.1.0` without adding new features.

**Comment by ColonelPanicks** [comment]
I can see your reasoning for it not being considered a bug because the node could be manually added but seeing as it is configured and built with metalware I think it should be completely handled by metalware to keep the self node as consistent with regular nodes as possible. 

To expand on the solutions, they are:

1. Manually add the node to `/etc/hosts`
_Happy to have this as a workaround for now but for the reasons mentioned above I think it should be handled by metalware._

2. Add the self node to a group so that it is included in the current loop in `hosts/default`
_This would be fine except that the hostname for the self node does not line up with `alces.nodename` so the line:_ 
    ```
    <%= network.ip %> <%= network.hostname  %> <%= network.short_hostname %> <%= network.primary ? node.alces.nodename : '' %>
    ```
    _ would need to be changed to:_
    ```
    ```




The self node has a few differences from regular nodes that I believe means that it should be handled separately from regular nodes in `/etc/hosts` (instead of being added to a group). For example, the self node sets `short_hostname` of the network interfaces to a hardcoded string instead of `alces.nodename` which requires different handling in `hosts/default` than regular nodes. (However we could just use the same method for regular nodes that the self node requires and add it into a group)  

Although the `alces.self` namespace would be helpful for the above reasons (and potentially for other uses in the future) I understand parking it while we get a release sorted out and (for now) manually adding the self node to `/etc/hosts`.

**Comment by WilliamMcCumstie** [comment]
Include `self` as node. Therefore `alces.nodes.self` should return the self node with `alces.self` as a shortcut.

The node namespace should contain a `self?` method that returns if the node is the self node.

**Comment by WilliamMcCumstie** [comment]
This has been renamed as `local`.

There is `alces.nodes.local` which is the local node. This is aliased to `alces.local`.

**Comment by WilliamMcCumstie** [comment]
This issue has been implemented in #227.

The local node now appears in the list of nodes (`alces.nodes.local`) and can be accessed directly with `alces.local`.
